File: apps/windows-client/src/ThemeRuntime.cs
namespace GoGate.WindowsClient;

public enum ThemeProfile
{
    Lite,
    Balanced,
    Rich
}

public record ThemeRuntimeSettings(string Animations, int FrameBudgetMs, int BackgroundPollMs, bool ReducedMotion);

public class ThemeRuntime
{
    public ThemeProfile CurrentProfile { get; private set; } = ThemeProfile.Balanced;
    public ThemeRuntimeSettings CurrentSettings { get; private set; } =
        new("standard", 16, 1200, false);

    public ThemeRuntimeSettings Apply(ThemeProfile profile, bool reducedMotion = false)
    {
        CurrentProfile = profile;
        CurrentSettings = profile switch
        {
            ThemeProfile.Lite => new ThemeRuntimeSettings("minimal", 16, 2000, reducedMotion),
            ThemeProfile.Rich => new ThemeRuntimeSettings("advanced", 16, 800, reducedMotion),
            _ => new ThemeRuntimeSettings("standard", 16, 1200, reducedMotion),
        };
        return CurrentSettings;
    }
}

File: apps/android-client/src/ThemeRuntime.kt
package com.gogate.android

enum class ThemeProfile { LITE, BALANCED, RICH }

data class ThemeRuntimeSettings(
    val animations: String,
    val frameBudgetMs: Int,
    val backgroundPollMs: Int,
    val reducedMotion: Boolean
)

class ThemeRuntime {
    var currentProfile: ThemeProfile = ThemeProfile.BALANCED
        private set

    var currentSettings: ThemeRuntimeSettings =
        ThemeRuntimeSettings("standard", 16, 1200, false)
        private set

    fun apply(profile: ThemeProfile, reducedMotion: Boolean = false): ThemeRuntimeSettings {
        currentProfile = profile
        currentSettings = when (profile) {
            ThemeProfile.LITE -> ThemeRuntimeSettings("minimal", 16, 2000, reducedMotion)
            ThemeProfile.RICH -> ThemeRuntimeSettings("advanced", 16, 800, reducedMotion)
            ThemeProfile.BALANCED -> ThemeRuntimeSettings("standard", 16, 1200, reducedMotion)
        }
        return currentSettings
    }
}
